chaosazure/common/compute/command.py

```diff
@@ -20,42 +20,42 @@ def prepare_path(machine: dict, path: str):
     return result
 
 
-def prepare(machine: dict, script: str):
-    _os_type = __get_os_type(machine)
-    if _os_type == OS_LINUX:
-        _command_id = 'RunShellScript'
-        _script_name = "{}.sh".format(script)
+def prepare(compute: dict, script: str):
+    os_type = __get_os_type(compute)
+    if os_type == OS_LINUX:
+        command_id = 'RunShellScript'
+        script_name = "{}.sh".format(script)
     else:
         if script in UNSUPPORTED_WINDOWS_SCRIPTS:
             raise InterruptExecution("'{}' is not supported for os '{}'"
                                      .format(script, OS_WINDOWS))
-        _command_id = 'RunPowerShellScript'
-        _script_name = "{}.ps1".format(script)
+        command_id = 'RunPowerShellScript'
+        script_name = "{}.ps1".format(script)
 
-    _file_path = os.path.join(
-        os.path.dirname(__file__), "../scripts", _script_name)
-    with open(_file_path) as _file_path:
-        _script_content = _file_path.read()
-        return _command_id, _script_content
+    file_path = os.path.join(
+        os.path.dirname(__file__), "../scripts", script_name)
+    with open(file_path) as file_path:
+        script_content = file_path.read()
+        return command_id, script_content
 
 
-def run(machine: dict, timeout: int, parameters: dict,
+def run(resource_group: str, compute: dict, timeout: int, parameters: dict,
```

Comment:
Why do we have to pass resource_group? Are we removing resource group from compute object somwhere before?

Reply:
It's not inside every compute object. Virtual machine do have them, Scale set instances do not have them. So I am passing this attribute explicitly.

I guess that Availability Sets and other Azure compute sets will be treated the same way.